engine/collection/message_hub/message_hub.go

```diff
@@ -0,0 +1,473 @@
+package message_hub
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/consensus/hotstuff"
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/consensus/hotstuff/notifications"
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/engine/common/fifoqueue"
+	"github.com/onflow/flow-go/model/events"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/flow/filter"
+	"github.com/onflow/flow-go/model/messages"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/network"
+	"github.com/onflow/flow-go/network/channels"
+	clusterkv "github.com/onflow/flow-go/state/cluster"
+	"github.com/onflow/flow-go/state/protocol"
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/utils/logging"
+)
+
+// packedVote is a helper structure to pack recipientID and vote into one structure to pass through fifoqueue.FifoQueue
+type packedVote struct {
+	recipientID flow.Identifier
+	vote        *messages.ClusterBlockVote
+}
+
+// MessageHub is a central module for handling incoming and outgoing messages via cluster consensus channel.
+// It performs message routing for incoming messages by matching them by type and sending to respective engine.
+/* For incoming messages handling processing looks like this:
```

Comment:
This is specifically for gofmt not to format this section

Reply:
I've tried different ways, if you know a better one please be me guest :)

Reply:
You can include diagrams the same way as you would include code (see `Code blocks` section in [Go Doc Comments](https://go.dev/doc/comment)) 
* have a blank line between the text and the diagram
* add 3 whitespaces in front of every line of the diagram
* (add a blank line between the diagram and the following text, if there is any)

Reply:
The following works on my IDE for [collector's `MessageHub`](https://github.com/onflow/flow-go/blob/b8627ddf7389dc5498ed465777c53b88173c4c25/engine/collection/message_hub/message_hub.go#L49-L75)
```golang
// MessageHub is a central module for handling incoming and outgoing messages via cluster consensus channel.
// It performs message routing for incoming messages by matching them by type and sending to respective engine.
// For incoming messages handling processing looks like this:
//
//      +-------------------+      +------------+
//   -->|  Cluster-Channel  |----->| MessageHub |
//      +-------------------+      +------+-----+
//                            ------------|------------
//      +------+---------+    |    +------+-----+     |    +------+------------+
//      | VoteAggregator |----+    | Compliance |     +----| TimeoutAggregator |
//      +----------------+         +------------+          +------+------------+
//             vote                     block                  timeout object
//
// MessageHub acts as communicator and handles hotstuff.Consumer communication events to send votes, broadcast timeouts
// and proposals. It is responsible for communication between cluster consensus participants.
// It implements hotstuff.Consumer interface and needs to be subscribed for notifications via pub/sub.
// All communicator events are handled on worker thread to prevent sender from blocking.
// For outgoing messages processing logic looks like this:
//
//    +-------------------+      +------------+      +----------+      +------------------------+
//    |  Cluster-Channel  |<-----| MessageHub |<-----| Consumer |<-----|        Hotstuff        |
//    +-------------------+      +------+-----+      +----------+      +------------------------+
//                                                      pub/sub          vote, timeout, proposal
//
// MessageHub is safe to use in concurrent environment.
type MessageHub struct {
```